internal/assemble/assemble.go

```diff
@@ -29,6 +29,36 @@ func assembleInput(ti typeNameToInfo, p *parse.InputPart) error {
 	return fmt.Errorf("unknown type: %s", p.Source.Prefix)
 }
 
+type sourceField int
+
+const (
+	fromString sourceField = iota
+	fromName
+)
+
+// createOutCols creates the list of output columns from the parameters passed
+// in.
+// The source and sf define the field to use from the FullName. If inf is not
+// nil, then it is used to run some checks.
+// Returns the array of columns and nil on success, nil and an error otherwise.
+func createOutCols(source []parse.FullName, sf sourceField, inf *typeinfo.Info) ([]string, error) {
+	var outCols []string
+	for _, c := range source {
+		if inf != nil {
+			if _, ok := inf.TagToField[c.Name]; !ok {
+				return nil, fmt.Errorf("there is no tag with name %s in %s",
+					c.Name, inf.Type.Name())
+			}
+		}
+		if sf == fromString {
+			outCols = append(outCols, c.String())
+		} else {
```

Comment:
Maybe put a comment saying that this is the `fromName` case, or even change it to an `else if` to make it slightly more readable.